src/common/config/RegistryConfig.js

```diff
@@ -131,22 +122,24 @@ export default class RegistryConfig extends EventEmitter {
    * @param {string} name Name of the specific entry to retrieve (optional)
    */
   getRegistryEntryValues(hive, key, name) {
-    return new Promise((resolve, reject) => {
+    const registry = new WindowsRegistry({hive, key});
+    return new Promise((resolve) => {
       try {
-        const results = registry.enumerateValues(hive, key);
-        if (!results || results.length === 0) {
-          resolve();
-          return;
-        }
-        if (name) { // looking for a single entry value
-          const registryItem = results.find((item) => item.name === name);
-          resolve(registryItem && registryItem.value ? registryItem.value : null);
-        } else { // looking for an entry list
-          resolve(results);
-        }
+        registry.values((error, results) => {
+          if (error || !results || results.length === 0) {
+            resolve();
+            return;
+          }
+          if (name) { // looking for a single entry value
+            const registryItem = results.find((item) => item.name === name);
+            resolve(registryItem && registryItem.value ? registryItem.value : null);
+          } else { // looking for an entry list
+            resolve(results);
+          }
+        });
       } catch (e) {
         log.error(`There was an error accesing the registry for ${name}: ${e}`);
-        reject(e);
```

Comment:
How come we aren't using the usual failure mode of `Promise` here anymore?

Reply:
originally it wasn't using the resolve/reject, only resolve. Since I was going back, i also went back to that. I think I prefer to use the resolve/reject mechanism, but since it was kind of manual revert I went back to the original.

if you prefer the resolve/reject I'll add it back